### PR TITLE
translation macro placeline switch: apply the french geonames value also for AP ingested stories [SDCP-560]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

File: server/cp/ingest/parser/ap.py [hunk 1]
```diff
@@ -90,6 +91,13 @@ def process_headline(self, headline):
             .replace("_", " - ")
         )
 
+    def set_dateline_date(self, ap_item):
```
Comment (petrjasek) [inline server/cp/ingest/parser/ap.py:94]: it starts with `set` but it's rather `get`

File: server/cp/macros/update_translation_metadata_macro.py [hunk 2]
```diff
@@ -52,12 +55,51 @@ def get_destination(items, qcode):
...
+            pass
+    return item
+
+
 def update_translation_metadata_macro(item, **kwargs):
     req = ParsedRequest()
     req.args = {}
 
-    located = item.get("dateline", {}).get("located")
+    item = set_dateline_for_translation(item)
```
Comment (petrjasek) [inline server/cp/macros/update_translation_metadata_macro.py:100]: not sure if `set` method in general should return anything, it's a bit confusing

File: server/tests/macros/update_translation_metadata_macro_test.py [hunk 3]
```diff
@@ -143,3 +148,54 @@ def test_destination_is_empty_add_presse_canadienne_staff(self):
...
+        }
+
+        with self.app.app_context():
+            with patch.dict(superdesk.resources, resources):
+                macro(item)
+
+        dateline = item.get("dateline")
+        self.assertEqual("The Associated Press", dateline["source"])
+        self.assertIn("located", dateline)
+        self.assertEqual("Torontoq", dateline["located"]["city"])
```
Comment (petrjasek) [inline server/tests/macros/update_translation_metadata_macro_test.py:197]: would be good to use some place where the translation would be actually different, here the values from mock and original are the same besides the typo so it's hard to tell if the macro does anything

Comment (GyanP) [reply server/tests/macros/update_translation_metadata_macro_test.py:197]: yep, you are right I will do changes.

File: server/cp/macros/update_translation_metadata_macro.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ def update_translation_metadata_macro(item, **kwargs):
     req = ParsedRequest()
     req.args = {}
 
-    item = set_dateline_for_translation(item)
+    item = get_dateline_for_translation(item)
```
Comment (petrjasek) [inline server/cp/macros/update_translation_metadata_macro.py:100]: you actually set it there, so it should be imo either:
```py
item["dateline"] = get_dateline_for_translation(item)
```
or
```py
set_dateline_for_translation(item)
```

File: server/cp/ingest/parser/ap.py [hunk 5]
```diff
@@ -667,7 +676,7 @@ def get_index(mapping):
         return ["Spare News"]
 
     def _parse_genre(self, data, item):
-        """VersionType in JIMI"""
+        """Versiontype in JIMI"""
```
Comment (petrjasek) [inline server/cp/ingest/parser/ap.py:679]: :eyes:

**Review by petrjasek — APPROVED** [review]

**Pull request merged**